plugins/none-ls/sources.nix

```diff
@@ -0,0 +1,22 @@
+{ config, lib, ... }:
+let
+  noneLsBuiltins = import ../../generated/none-ls.nix;
+  mkSourcePlugin = import ./_mk-source-plugin.nix;
+in
+{
+  imports =
+    [ ./prettier.nix ]
+    ++ (lib.flatten (
+      lib.mapAttrsToList (category: (lib.map (mkSourcePlugin category))) noneLsBuiltins
+    ));
+
+  config =
+    let
+      cfg = config.plugins.none-ls;
+      gitsignsEnabled = cfg.sources.code_actions.gitsigns.enable;
+    in
+    lib.mkIf cfg.enable {
+      # Enable gitsigns if the gitsigns source is enabled
+      plugins.gitsigns.enable = lib.mkIf gitsignsEnabled true;
```

Comment:
Do you think there is a way to have a warning if gitsigns was not already enabled? I think we will end up with an infinite recursion, but maybe you know a way to to that?
If we can add a warning we could remove this magic in 24.11

Reply:
I think this is possible by checking the `options.plugins.gitsigns.enable.definitions` length.

If the definitions only has one element, then that's our mkIf. If there's multiple definitions then the user has defined it themselves.

Would we also want the warning to check if the final merged value would be `true` as well?

Reply:
While I think it's possible, I don't really want to tackle it in this PR.

Reply:
Yeah no problem